ftplugin/c.lua

```diff
@@ -1,3 +1,18 @@
+local clang_formatter = {
+  function()
+    return {
+      exe = "clang-format",
+      --  TODO: append to this for args don't overwrite
+      args = {},
+    }
+  end,
+}
+
+O.formatters.filetype["c"] = clang_formatter
+O.formatters.filetype["cpp"] = clang_formatter
+O.formatters.filetype["cmake"] = clang_formatter
```

Comment:
I don't think clang-format handles cmake formatting, however, I use [cmake-format](https://github.com/cheshirekow/cmake_format) and it works great

Reply:
nice catch 👀 i'm adding correct changes rn